python/tvm/relay/op/contrib/arm_compute_lib.py

```diff
@@ -289,24 +293,57 @@ def qnn_dense(attrs, args):
         return False
     if attrs.out_dtype != "int32":
         return False
-    return True
+
+    return not padding_required([*args, out_type])
 
 
 @tvm.ir.register_op_attr("nn.max_pool2d", "target.arm_compute_lib")
-def max_pool2d(attrs, args):
+def max_pool2d(attrs, args, out_type):
     """Check if the external ACL codegen for maxpool2d should be used."""
     if attrs.layout != "NHWC":
         return False
     typ = args[0].checked_type
     if typ.dtype not in ["float32", "uint8"]:
         return False
-    return True
+    return not padding_required([*args, out_type])
+
+
+def padding_required(inputs):
```

Comment:
- `require_padding` might be more straightforward?
- I feel that this checker can be more general to support both versions. For example, you can check the ACL version and always return True if it's < 20.05. In this way, this PR won't be a "temporary" fix anymore.
- If we simply require ACL 20.11 or higher, can we get rid of this problem without any change?

Reply:
>If we simply require ACL 20.11 
ACL 20.11 is not available now, and will be ready approximately in 6 weeks.

Reply:
A fix for approximately 6 weeks should not be "temporary" IMHO...

Reply:
>require_padding might be more straightforward?
renamed

Agreed, all the current versions of ACL should use the fix, until migration to ACL 20.11 (when it became available)